frontend/src/v5/ui/routes/dashboard/projects/containers/createContainerForm/createContainerForm.component.tsx

```diff
@@ -0,0 +1,156 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import { formatMessage } from '@/v5/services/intl';
+import { FormattedMessage } from 'react-intl';
+import { TextField, MenuItem } from '@material-ui/core';
+import { FormModal } from '@/v5/ui/controls/modal/formModal/formDialog.component';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { useParams } from 'react-router';
+import { ContainersActionsDispatchers } from '@/v5/services/actionsDispatchers/containersActions.dispatchers';
+import { Select } from './createContainerForm.styles';
+
+interface IFormInput {
+	name: string;
+	unit: string;
+	desc: string;
+	code: string;
+	type: string;
+}
+
+const errorMessage = {
+	name: {
+		required: formatMessage({ id: 'containers.creation.name.error.required', defaultMessage: 'Name is a required field' }),
+		maxLength: formatMessage({ id: 'containers.creation.name.error.length', defaultMessage: 'Name is limited to 120 characters' }),
+	},
+	code: {
+		maxLength: formatMessage({ id: 'containers.creation.code.error.length', defaultMessage: 'Code is limited to 50 characters' }),
+		pattern: formatMessage({ id: 'containers.creation.code.error.pattern', defaultMessage: 'Code is limited to letters and numbers' }),
+	},
```

Comment:
I think we could use a more general system for the messages. Something of the sort `formatMessage({ id: 'forms.validation.required', defaultMessage: '{fieldName} is limited to {maxLength} characters.' }), { fieldName: formatMessage(id: 'containers.create.codeFieldName'...), maxLength: 50} )` .

I dont know exactly the specifics of how to change that with yup, but here is an example. Maybe it could be implemented for all required field. You need to research a bit:
https://github.com/jquense/yup/issues/47#issuecomment-215588412